packages/terra-application-utility/src/utility/_UtilityMenu.jsx

```diff
@@ -0,0 +1,319 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import Button from 'terra-button';
+import ContentContainer from 'terra-content-container';
+import IconClose from 'terra-icon/lib/icon/IconClose';
+import IconLeft from 'terra-icon/lib/icon/IconLeft';
+import MenuDivider from './_UtilityMenuDivider';
+import Utils from '../_Utils';
+import MenuItem from './_UtilityMenuItem';
+import styles from './_UtilityMenu.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The initial selected key. Used as the top level menu page.
+   */
+  selectedKey: PropTypes.string.isRequired,
+  /**
+   * Indicates if the height is bound to it's parent container.
+   */
+  isHeightBounded: PropTypes.bool,
+  /**
+   * The array of the menu items configs to be rendered.
+   */
+  menuItems: PropTypes.arrayOf((Utils.itemShape)).isRequired,
+  /**
+   * The function to trigger when a menu item is selected.
+   */
+  onChange: PropTypes.func.isRequired,
+  /**
+   * The function that closes the menu.
+   * This will be provided by the terra-application-header or terra-application-menu
+   */
+  onRequestClose: PropTypes.func,
+  /**
+   * Sets the Utility variant. One of Utils.VARIANTS.HEADER, Utils.VARIANTS.MENU.
+   */
+  variant: PropTypes.oneOf([Utils.VARIANTS.HEADER, Utils.VARIANTS.MENU]),
+};
+
+const processMenuItems = (items) => {
+  const map = new Map();
+  items.forEach((item) => {
+    map.set(
+      item.key,
+      { id: item.id,
+        itemKey: item.key,
+        title: item.title,
+        content: item.content,
+        contentLocation: item.contentLocation,
+        isSelected: item.isSelected,
+        isSelectable: item.isSelectable,
+        childKeys: item.childKeys,
+      },
+    );
+  });
+  return map;
+};
+
+const hasChevron = item => item.childKeys && item.childKeys.length > 0;
+
+const contextTypes = {
```

Comment:
Can you use injectIntl to interface with the context? https://github.com/yahoo/react-intl/blob/master/src/inject.js#L19

Reply:
Switched to `injectIntl` for Utility Button and Menu [here](https://github.com/cerner/terra-framework/pull/103/commits/efd64c3eed5f7146aaefa0505339e34e92f75bed#diff-8de1146c08c79a8c3984e24f6e1ef89aR140) and [here](https://github.com/cerner/terra-framework/pull/103/commits/efd64c3eed5f7146aaefa0505339e34e92f75bed#diff-69f938b44661317549223f97d5ce876fR345).